### PR TITLE
Update defaults

### PR DESCRIPTION
**Pull request by ThomasKoppensteiner**
New cops are
```
Lint/RaiseException:
Lint/StructNewOverride:

Style/ExponentialNotation:
Layout/SpaceAroundMethodCallOperator:
```

All others were moved around to sort them (see commits).

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -81,40 +99,48 @@ Layout/MultilineOperationIndentation:
...
+#### Rest
+Style/Documentation:
+  Enabled: false
 
 Style/DoubleNegation:
   Enabled: false
 
-Style/RegexpLiteral:
+Style/ExponentialNotation:
+  EnforcedStyle: engineering
```
Comment (ThomasKoppensteiner) [inline config/default.yml:110]: https://docs.rubocop.org/en/latest/cops_style/#styleexponentialnotation `scientific` would be the default value.

File: rt_rubocop_defaults.gemspec [hunk 2]
```diff
@@ -22,7 +22,7 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
 
-  spec.add_dependency "rubocop", "~> 0.77"
+  spec.add_dependency "rubocop", "~> 0.81"
```
Comment (ThomasKoppensteiner) [inline rt_rubocop_defaults.gemspec:25]: Current version is `0.82`, but it dropped the support for ruby `2.3`.

**Comment by ahmgeek** [comment]
👏

File: config/default.yml [hunk 3]
```diff
@@ -12,16 +12,43 @@ Lint/NestedMethodDefinition:
...
+  Enabled: true
+
 ### Metrics
 
-Layout/LineLength:
+Metrics/AbcSize:
+  Max: 18
+  Severity: refactor
+  Exclude:
+    - api/sinatra/**/*
```
Comment (Goltergaul) [inline config/default.yml:27]: this is not fitting our shed structure. Whats the intention of this cop with the severity level? not sure from the docs what it does

File: config/default.yml [hunk 4]
```diff
@@ -12,16 +12,43 @@ Lint/NestedMethodDefinition:
...
+
+Metrics/BlockLength:
+  Exclude:
+    - '*.gemspec'
+    - '**/*.rake'
+    - 'config/initialize/*'
+    - 'spec/**/*'
+    - 'api/**/*'
```
Comment (Goltergaul) [inline config/default.yml:35]: this is not fitting the shed structure

File: config/default.yml (hunk 3 above)
Comment (ThomasKoppensteiner) [reply config/default.yml:27]: Good question. This cop was just moved up to have some kind of order in here (See commit message).

I'll update the description.

Comment (ThomasKoppensteiner) [reply config/default.yml:27]: Should we even assume any folder structure here?

Comment (andreaseger) [reply config/default.yml:27]: we can update this one with the shed structure. intention was just to not have rubocop complain about about the "complexity" of the big blocks in the api files

File: config/default.yml (hunk 4 above)
Comment (andreaseger) [reply config/default.yml:35]: @ThomasKoppensteiner please update with the relevant shed structures

File: config/default.yml (hunk 3 above)
Comment (Goltergaul) [reply config/default.yml:27]: ah cool did not pay attention to the fact that this code only moved sorry, but maybe we can add the shed path for this as well, then it should be fine :)

File: rt_rubocop_defaults.gemspec (hunk 2 above)
Comment (Goltergaul) [reply rt_rubocop_defaults.gemspec:25]: i guess we could still update to 82, ruby 2.3 is not supported anymore anyways. old projects can stick to the old rt_rubocop_defaults i guess?

Comment (andreaseger) [reply rt_rubocop_defaults.gemspec:25]: well the defaults are forward compatible aren't the?

Comment (ThomasKoppensteiner) [reply rt_rubocop_defaults.gemspec:25]: I'll suggest doing the upgrade (and the breaking change) to `0.82` in a new PR.
With that we'll also can remove some renamed cops etc.

**Pull request merged**